deepchecks/base/check.py

```diff
@@ -332,6 +336,67 @@ def name(cls):
         """Name of class in split camel case."""
         return split_camel_case(cls.__name__)
 
+    # NOTE: next set of private functions exists to unify error messages across all checks
```

Comment:
General note - 
One more error that is currently a "DeepchecksValueError" is in the BoostingOverfit check... 
(if model is not of type boosting. not sure where this should be but it also belongs to these types of "irrelevant data/model for check")